### PR TITLE
Support the ability to disable DR

### PR DESCRIPTION
**Pull request by TimothyAsirJeyasing**

### CODE REVIEW

**Comment by SanjalKatiyar** [comment]
@TimothyAsirJeyasing add screenshots...
@GowthamShanmugam do we want to add unit tests as well ??

**Comment by GowthamShanmugam** [comment]
@SanjalKatiyar  @TimothyAsirJeyasing  Yes, from 4.14, lets start adding unit test for all new code

**Comment by TimothyAsirJeyasing** [comment]
https://github.com/red-hat-storage/odf-console/assets/6670284/f4f50c54-0b4f-48e6-ab8b-dd355c0c94cc

**Comment by TimothyAsirJeyasing** [comment]
During disable
Screenshot from 2023-05-29 14-36-49
After disable:

**Comment by TimothyAsirJeyasing** [comment]
After disable
Screenshot from 2023-05-16 18-37-33

**Comment by TimothyAsirJeyasing** [comment]
After Disable
Screenshot from 2023-05-29 14-37-49

File: packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx [hunk 1]
```diff
@@ -120,29 +119,15 @@ const ApplyDRPolicyModal: React.FC<CommonModalProps<ApplyModalExtraProps>> = (
...
         type: ApplyPolicyType.SET_MESSAGE,
         payload: {
-          text: t('Cannot unassign the policy'),
-          description: t(
-            'You cannot unassign the policy as this action is not yet supported.'
+          text: t(
+            'Disabling DR protection for applications will no longer protect the underlying PVCs. You must reapply the policy in order to enable DR protection for your application.'
           ),
-          variant: AlertVariant.info,
-          type: MessageType.UNSUPPORTED_OPERATION,
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx:128]: lets keep variant and type on type declaration, no need to remove it completely

File: packages/mco/components/modals/drpolicy-apps-apply/reducer.ts [hunk 2]
```diff
@@ -2,15 +2,9 @@ import { AlertVariant } from '@patternfly/react-core';
...
-export enum MessageType {
-  UNSUPPORTED_OPERATION,
-}
-
 type Message = {
   text: string;
-  description?: string;
-  type?: MessageType;
-  variant: AlertVariant;
+  type: AlertVariant;
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/drpolicy-apps-apply/reducer.ts:7]: no need to remove anything

File: packages/mco/components/modals/drpolicy-apps-apply/reducer.ts [hunk 3]
```diff
@@ -2,15 +2,9 @@ import { AlertVariant } from '@patternfly/react-core';
...
 };
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/drpolicy-apps-apply/reducer.ts:8]: Just remove:
type?: MessageType;

Comment (TimothyAsirJeyasing) [reply packages/mco/components/modals/drpolicy-apps-apply/reducer.ts:14]: Done

**Comment by SanjalKatiyar** [comment]
> @SanjalKatiyar @TimothyAsirJeyasing Yes, from 4.14, let's start adding unit tests for all new code, Ivo can help

@TimothyAsirJeyasing add unit tests for this PR.

File: packages/mco/components/modals/drpolicy-apps-apply/reducer.ts [hunk 4]
```diff
@@ -9,8 +9,8 @@ export enum MessageType {
 type Message = {
   text: string;
   description?: string;
-  type?: MessageType;
   variant: AlertVariant;
+  type?: MessageType;
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/drpolicy-apps-apply/reducer.ts:13]:
```suggestion

```

File: packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx [hunk 5]
```diff
@@ -120,29 +120,16 @@ const ApplyDRPolicyModal: React.FC<CommonModalProps<ApplyModalExtraProps>> = (
...
         payload: {
-          text: t('Cannot unassign the policy'),
-          description: t(
-            'You cannot unassign the policy as this action is not yet supported.'
+          text: t(
+            'Disabling DR protection for applications will no longer protect the underlying PVCs. You must reapply the policy in order to enable DR protection for your application.'
           ),
-          variant: AlertVariant.info,
+          variant: AlertVariant.warning,
           type: MessageType.UNSUPPORTED_OPERATION,
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx:127]:
```suggestion
       
```

File: packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx [hunk 6]
```diff
@@ -120,29 +120,16 @@ const ApplyDRPolicyModal: React.FC<CommonModalProps<ApplyModalExtraProps>> = (
       dispatch({
         type: ApplyPolicyType.SET_MESSAGE,
         payload: {
-          text: t('Cannot unassign the policy'),
-          description: t(
-            'You cannot unassign the policy as this action is not yet supported.'
+          text: t(
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx:123]:
```suggestion
          description: t(
```

File: packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx [hunk 7]
```diff
@@ -282,9 +269,7 @@ const ApplyDRPolicyModal: React.FC<CommonModalProps<ApplyModalExtraProps>> = (
                 variant={state.message.variant}
                 title={state.message.text}
                 className="odf-alert odf-alert--margin-top"
-              >
-                {state.message?.description}
-              </Alert>
+              />
```
Comment (GowthamShanmugam) [inline packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.tsx:272]: This is generic, better keep {state.message?.description}

File: packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+}));
+
+jest.mock('@odf/shared/hooks/k8s-get-hook', () => ({
+  __esModule: true,
+  useK8sGet: jest.fn().mockReturnValue([undefined, true, undefined]),
+}));
+
+jest.mock('./reducer', () => ({
+  ApplyPolicyReducer: (state, action) => {
+    switch (action.type) {
```
Comment (ivoalmeida) [inline packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.spec.tsx:21]: @TimothyAsirJeyasing why are you re-implementing this hook here?

Comment (ivoalmeida) [reply packages/mco/components/modals/drpolicy-apps-apply/apply-policy-modal.spec.tsx:21]: Try returning the value that you need

**Comment by GowthamShanmugam** [comment]
This PR is not required anymore please close this.

**Pull request closed without merging**